### DISCUSSION TITLE
JSON command output

### DISCUSSION THREAD

**Discussion by sandronoel** [MusicPlayerDaemon/MPD#1723] [Ideas]
I'm writing a MPD Client framework for Swift.

While processing all the command results, I was wondering if anyone thought it would be a good idea to have a JSON response and query instead of the \n terminated strings? Before you light me up, I know nothing about modern c++'s magic sorry, I was good in c++ 25 years ago, else I'd fork and write it myself. ;) 

Is there a plugin that already does that and I missed it ?

The protocol documentation would also profit from having response to commands examples.

cheers !
Sandro.

**Comment by MaxKellermann** [comment]
MPD plugins cannot change the protocol. 
But I think you're asking the wrong question. I'm not convinced that adding JSON to the MPD protocol would be a good idea. Are you?

**Reply by sandronoel** [reply]
Thanks for relying so quickly!
Yes I'm suggesting that JSON is a way to transfer structured and typed replies in a string based protocol.

**Reply by MaxKellermann** [reply]
But MPD's existing protocol is already capable of transferring structured typed replies in a string based protocol. It seems you have no point other than proposing JSON "just because".

**Reply by sandronoel** [reply]
Great, 
Thank you for your time.
